subsys/bluetooth/mesh/access.c

```diff
@@ -456,25 +459,81 @@ struct bt_mesh_elem *bt_mesh_elem_find(uint16_t addr)
 {
 	uint16_t index;
 
+	if (!BT_MESH_ADDR_IS_UNICAST(addr)) {
+		return NULL;
+	}
+
+	index = addr - dev_comp->elem[0].addr;
+	if (index >= dev_comp->elem_count) {
+		return NULL;
+	}
+
+	return &dev_comp->elem[index];
+}
+
+bool bt_mesh_has_addr(uint16_t addr, bool trigger_msg_cb)
+{
+	uint16_t index;
+
 	if (BT_MESH_ADDR_IS_UNICAST(addr)) {
-		index = (addr - dev_comp->elem[0].addr);
-		if (index < dev_comp->elem_count) {
-			return &dev_comp->elem[index];
-		} else {
-			return NULL;
-		}
+		return bt_mesh_elem_find(addr) != NULL;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_MESH_ACCESS_LAYER_MSG) && trigger_msg_cb) {
+		return true;
 	}
 
 	for (index = 0; index < dev_comp->elem_count; index++) {
 		struct bt_mesh_elem *elem = &dev_comp->elem[index];
 
 		if (bt_mesh_elem_find_group(elem, addr)) {
-			return elem;
+			return true;
 		}
 	}
 
-	return NULL;
+	return false;
+}
+
+#if defined(CONFIG_BT_MESH_ACCESS_LAYER_MSG)
+void bt_mesh_msg_cb_set(void (*cb)(uint32_t opcode, struct bt_mesh_msg_ctx *ctx,
+			struct net_buf_simple *buf))
+{
+	msg_cb = cb;
+}
+
+int bt_mesh_msg_send(struct bt_mesh_msg_ctx *ctx, struct net_buf_simple *buf)
+{
+	struct bt_mesh_cdb_app_key *app_key;
+
+	if (!BT_MESH_IS_DEV_KEY(ctx->app_idx)) {
+		app_key = bt_mesh_cdb_app_key_get(ctx->app_idx);
+		if (!app_key) {
+			return -EINVAL;
+		}
+
+		ctx->net_idx = app_key->net_idx;
+	}
+
+	if (!bt_mesh_is_provisioned()) {
+		return -EAGAIN;
+	}
+
+	if (net_buf_simple_tailroom(buf) < 4) {
+		return -EINVAL;
+	}
+
+	if (buf->len > BT_MESH_TX_SDU_MAX - 4) {
+		return -EMSGSIZE;
+	}
+
+	struct bt_mesh_net_tx tx = {
+		.ctx = ctx,
+		.src = dev_primary_addr,
```

Comment:
If you make this source address a parameter for this function, you can make `bt_mesh_model_send` a small wrapper around this one.

Reply:
I'm pretty sure I understand what you meen here @trond-snekvik, but just to double check, you mean my function bt_mesh_msg_send could be a small wrapper around bt_mesh_model_send, where bt_mesh_msg_send just calls bt_mesh_model_send with the parameters cb and cb_data set to NULL, correct?

Reply:
Yep!

Reply:
Ok. So in this case, near the end of bt_mesh_model_send there is a check model_has_key, which isn't relevant to this use case. If I make some changes to bt_mesh_model_send so that it can be used by bt_mesh_msg_send without actually having a model structure, I'm not sure it's going to look any better than just leaving the two functions as they are. Thoughts?

Reply:
Whoops, looks like I misread your first question. It's time for bed, I think 😄

I meant that the wrapping would be the other way around - this new function is called by model send, which does the key binding check on the model

Reply:
Ohhhhhh kay. I see now. I can do this :)